src/utils.js

```diff
@@ -148,3 +148,49 @@ export function sanityCheck(
 
   return null;
 }
+
+export function getCryptoPassword(password: string) {
+  return `{SHA}'${crypto
+    .createHash('sha1')
+    .update(password, 'binary')
+    .digest('base64')}`;
+}
+
+/**
+ * changePasswordToHTPasswd - change password for existing user
+ * @param {string} body
+ * @param {string} user
+ * @param {string} passwd
+ * @param {string} newPasswd
+ * @returns {string}
+ */
+export function changePasswordToHTPasswd(
+  body: string,
+  user: string,
+  passwd: string,
+  newPasswd: string
+): string {
+  if (crypt3) {
+    passwd = crypt3(passwd);
+    newPasswd = crypt3(newPasswd);
+  } else {
+    passwd = getCryptoPassword(passwd);
+    newPasswd = getCryptoPassword(newPasswd);
+  }
+
+  let lines = body.split('\n');
+  lines = lines.map(line => {
+    let [username, password] = line.split(':', 3);
+    if (username == user) {
+      if (password == passwd) {
```

Comment:
Won't they be of the same type?

Reply:
good one, I wonder why eslint does not catch this.